### PR TITLE
Support for u64

### PR DESCRIPTION
**Pull request by evanxg852000**
### Context / purpose
Closes https://github.com/quickwit-inc/quickwit/issues/351

### Description
- Implemented few changes to support u64 type

### How was this PR tested?
Used with the CLI command to create, index, and search data

### Checklist
- [x] tested locally
- [x] added unit tests
- [x] included documentation

### CODE REVIEW

File: docs/reference/index-config.md [hunk 1]
```diff
@@ -91,8 +91,8 @@ Example of a mapping for a text field:
 | `position`    |  Records the document id, the term frequency and the positions of occurences  |
 
 
-#### Numeric types: `i64` and `f64` type
-Quickwit handles two numeric types: `i64` and `f64`.
+#### Numeric types: `i64`, `u64` and `f64` type
+Quickwit handles two numeric types: `i64`, `u64` and `f64`.
```
Comment (fulmicoton) [inline docs/reference/index-config.md:95]: three numeric types

**Comment by evanxg852000** [comment]
I tested with and index config with u64. will run through again

File: docs/reference/index-config.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ This config can be expressed as a json file given to the `new` Quickwit command.
 
 ## Field types
 Each field has a type which indicates the kind of data it contains such as integer on 64 bits or text.
-Quickwit supports the following raw types `text`, `i64`, `f64`, `date` and `bytes` and also supports composite types such as array and object. Behind the scenes, Quickwit is using tantivy field types, don't hesitate to have a look at [tantivy documentation](https://github.com/tantivy-search/tantivy) if you want to go into the details.
+Quickwit supports the following raw types `text`, `i64`, `u64`, `f64`, `date` and `bytes` and also supports composite types such as array and object. Behind the scenes, Quickwit is using tantivy field types, don't hesitate to have a look at [tantivy documentation](https://github.com/tantivy-search/tantivy) if you want to go into the details.
```
Comment (fmassot) [inline docs/reference/index-config.md:50]: nice :)))))))
never forget the docs!!

**Review by fmassot — APPROVED** [review]

File: quickwit-index-config/src/default_index_config/field_mapping_type.rs [hunk 3]
```diff
@@ -32,6 +32,8 @@ pub enum FieldMappingType {
     Text(TextOptions, Cardinality),
     /// Signed 64-bits integers mapping type configuration.
     I64(IntOptions, Cardinality),
+    /// Unsigned 64-bits integers mapping type configuration.
```
Comment (guilload) [inline quickwit-index-config/src/default_index_config/field_mapping_type.rs:35]: FYI, for next time: `64-bit` ([hyphen in compound adjective with numbers](https://www.grammarly.com/blog/hyphen-in-compound-adjective-with-numbers/)).

**Pull request merged**